org.eclipse.xtext/src/org/eclipse/xtext/resource/XtextResource.java

```diff
@@ -308,7 +310,17 @@ protected void updateInternalState(IParseResult newParseResult) {
 	boolean containsRootElement(EObject newRootASTElement) {
 		return getContents().contains(newRootASTElement);
 	}
-	
+
+	/**
+	 * @since 2.26
+	 */
+	protected void clearCache() {
+		IResourceScopeCache cache = getCache();
+		if (cache instanceof OnChangeEvictingCache) {
+			cache.clear(this);
```

Comment:
unfortunately this one calls getOrCreate

Reply:
Should we make `OnChangeEvictingCache.clear` smarter in that regard?

Reply:
i will have a look

Reply:
@szarnekow  is having a custom cache adapter a usecase?
if i dont call getOrCreate then customoizations of that one may break.

Reply:
I've not seen a custom cache yet. A since-tag should be fine. The contract of clear is, that there is no cached entry afterwards, so specializing that one such that the adapter is not installed if it wasn't present should be fine